cmd/crictl/main.go

```diff
@@ -55,17 +60,14 @@ func getRuntimeClientConnection(context *cli.Context) (*grpc.ClientConn, error)
 	if RuntimeEndpoint == "" {
 		return nil, fmt.Errorf("--runtime-endpoint is not set")
 	}
-
-	addr, dialer, err := util.GetAddressAndDialer(RuntimeEndpoint)
-	if err != nil {
-		return nil, err
-	}
-
-	conn, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithTimeout(Timeout), grpc.WithContextDialer(dialer))
-	if err != nil {
-		return nil, errors.Wrap(err, "connect, make sure you are running as root and the runtime has been started")
+	logrus.Debugf("get runtime connection")
+	// As dockershim is deprecated as a default CRI server, it still needs to
+	// be supported. This allows other default endpoint types to be
+	// checked if cannot connect to dockershim
+	if !RuntimeEndpointIsSet {
```

Comment:
won't get here see line 60 above.

Reply:
maybe use =="" as the bool

Reply:
It will get there because it uses a default value if not set:  https://github.com/kubernetes-sigs/cri-tools/blob/master/cmd/crictl/main.go#L190. The check for empty in line 60 is incase empty value set by cli or file.

Reply:
`if !RuntimeEndpointIsSet` should be a good check for bool false?

Reply:
if they set "" are they looking for an error? If it's not "" how is it not set? I'm confused ;-)

Reply:
yes, logic is a bit funny here now with the changes. I have refactored to hopefully improve it.